src/lang/cpp/internal/Parser.java

```diff
@@ -461,11 +461,7 @@ public ISet getMethodOverrides(IASTTranslationUnit tu) {
 
 	public ISet getMacroDefinitionsFromTranslationUnit(IASTTranslationUnit tu) {
 		return Stream.of(tu.getMacroDefinitions()).map(it -> {
-			try {
-				return vf.tuple(br.resolveBinding(it.getName().resolveBinding()), getSourceLocation(it));
-			} catch (URISyntaxException e) {
-				return vf.tuple(vf.sourceLocation(URIUtil.rootScheme("unknown")), getSourceLocation(it));
-			}
+			return vf.tuple(br.resolveBinding(it.getName().resolveBinding(), getSourceLocation(it)), getSourceLocation(it));
```

Comment:
Looks like `BindingsResolver::resolveBinding` could do with an extra overload (`IASTNameOwner` instead of `ISourceLocation`)

Reply:
that could help to make it shorter yes. good refactoring idea!